hail/src/main/scala/is/hail/expr/ir/ExtractIntervalFilters.scala

```diff
@@ -1,11 +1,32 @@
 package is.hail.expr.ir
 
+import is.hail.annotations.IntervalEndpointOrdering
 import is.hail.expr.types.virtual._
 import is.hail.utils.{FastSeq, Interval, IntervalEndpoint, _}
 import is.hail.variant.{Locus, ReferenceGenome}
 import org.apache.spark.sql.Row
 
 object ExtractIntervalFilters {
+
+  case class ExtractionState(rowRef: Ref, keyFields: IndexedSeq[String]) {
```

Comment:
This class seems to be assuming that `keyFields` is a prefix of the list of fields of `rowRef.typ`. Is that correct?

Reply:
no -- they're fields that appear anywhere in rowRef.typ, and in any order.

The isKeyStructPrefix rule is wrong (rowType should be rowKeyType), I fix this in the stacked PR. I can just remove that method here.

Reply:
Then you also need
```
val firstKeyType: Type = rowKeyType.types.head
def isFirstKey(ir: IR): Boolean = ir == GetField(rowRef, rowKeyType.fieldNames.head)
```

Reply:
yeah, saw that too. Will add a test that catches it.